test/tests/CMakeLists.txt

```diff
@@ -1,163 +1,39 @@
-endif ()
-
-if (NOT DISABLE_RCT2_TESTS)
-    execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink "${CMAKE_CURRENT_LIST_DIR}/testdata" "${CMAKE_CURRENT_BINARY_DIR}/testdata")
-endif ()
-
-# Start of our tests
-
-# sawyercoding test
-
-set(SAWYERCODING_TEST_SOURCES
-        "${CMAKE_CURRENT_LIST_DIR}/sawyercoding_test.cpp"
-        "${ROOT_DIR}/src/openrct2/diagnostic.c"
-        "${ROOT_DIR}/src/openrct2/util/sawyercoding.c"
-        "${ROOT_DIR}/src/openrct2/localisation/utf8.c"
-        )
-add_executable(test_sawyercoding ${SAWYERCODING_TEST_SOURCES})
-target_link_libraries(test_sawyercoding ${GTEST_LIBRARIES})
-add_test(NAME sawyercoding COMMAND test_sawyercoding)
-
-# LanguagePack test
-set(LANGUAGEPACK_TEST_SOURCES
-        "${CMAKE_CURRENT_LIST_DIR}/LanguagePackTest.cpp"
-        "${ROOT_DIR}/src/openrct2/localisation/LanguagePack.cpp"
-        )
-add_executable(test_languagepack ${LANGUAGEPACK_TEST_SOURCES})
-if (UNIX AND NOT ${CMAKE_SYSTEM_NAME} MATCHES "BSD")
-    # Include libdl for dlopen
-    set(LDL dl)
-endif ()
-target_link_libraries(test_languagepack ${GTEST_LIBRARIES} test-common ${LDL} z SDL2)
-add_test(NAME languagepack COMMAND test_languagepack)
-
-# INI test
-set(INI_TEST_SOURCES
-        "${CMAKE_CURRENT_LIST_DIR}/IniWriterTest.cpp"
-        "${CMAKE_CURRENT_LIST_DIR}/IniReaderTest.cpp"
-        "${ROOT_DIR}/src/openrct2/config/IniReader.cpp"
-        "${ROOT_DIR}/src/openrct2/config/IniWriter.cpp"
-        "${ROOT_DIR}/src/openrct2/core/IStream.cpp"
-        "${ROOT_DIR}/src/openrct2/core/MemoryStream.cpp"
-        )
-add_executable(test_ini ${INI_TEST_SOURCES})
-target_link_libraries(test_ini ${GTEST_LIBRARIES} test-common ${LDL} z)
-add_test(NAME ini COMMAND test_ini)
-
-# String test
-set(STRING_TEST_SOURCES
-        "${CMAKE_CURRENT_LIST_DIR}/StringTest.cpp"
-        )
-add_executable(test_string ${STRING_TEST_SOURCES})
-target_link_libraries(test_string ${GTEST_LIBRARIES} test-common ${LDL} z)
-add_test(NAME string COMMAND test_string)
-
-
-# Ride ratings test
-set(RIDE_RATINGS_TEST_SOURCES "${CMAKE_CURRENT_LIST_DIR}/RideRatings.cpp"
-                              "${CMAKE_CURRENT_LIST_DIR}/TestData.cpp")
-add_executable(test_ride_ratings ${RIDE_RATINGS_TEST_SOURCES})
-target_link_libraries(test_ride_ratings ${GTEST_LIBRARIES} libopenrct2 ${LDL} z)
-
-# Multi-launch test
-set(MULTILAUNCH_TEST_SOURCES "${CMAKE_CURRENT_LIST_DIR}/MultiLaunch.cpp"
-                             "${CMAKE_CURRENT_LIST_DIR}/TestData.cpp")
-add_executable(test_multilaunch ${MULTILAUNCH_TEST_SOURCES})
-target_link_libraries(test_multilaunch ${GTEST_LIBRARIES} libopenrct2 ${LDL} z)
-    
-if (NOT DISABLE_RCT2_TESTS)
-    add_test(NAME ride_ratings COMMAND test_ride_ratings)
-    add_test(NAME multilaunch COMMAND test_multilaunch)
-endif ()
+    set(ENABLE_RCT2_TESTS OFF)
+endif()
+
+function(add_orct2_test name)
+    add_executable(test-${name} ${ARGN})
+    target_link_libraries(test-${name} PRIVATE openrct2::core ${GTEST_LIBRARIES})
```

Comment:
This assumes all tests link to `libopenrct2`, I specifically wanted to avoid that to ensure that parts of the project are standalone.

Reply:
I fully support the idea of this on the other hand.

Reply:
That's why you did it so for msbuild and I did the other way for CMake.